components/search-modal/element.js

```diff
@@ -105,7 +106,9 @@ export class MDNSearchModal extends L10nMixin(LitElement) {
           );
           // Workaround: For some reason, the dispatched click
           // does not cause Glean.js to capture it via `data-glean-id`.
-          gleanClick(`quick-search: keyboard -> ${this._query}`);
+          if (item.dataset.gleanId) {
+            gleanClick(item.dataset.gleanId);
+          }
```

Comment:
Add `composed: true` to the `MouseEvent` above (so it escapes the shadow dom to our document event listener) and we can remove this workaround